File: level_2/구명보트/devSOWON0628.js
function solution(people, limit) {
    people.sort((a,b)=>(b-a))
    let l = 0
    let r = people.length -1
    let answer = 0
    while(l < r){
        if(people[l] + people[r] <= limit){
            l++
            r--
        }else{
            l++
        }
        if(l==r){
            answer++
        }
        answer++
    }    
    return answer;
}

File: level_2/괄호_회전하기/devSOWON0628.py
def solution(s):
    answer = 0
    n = len(s)
    match = { '(': ')', '{': '}', '[': ']' }

    for i in range(n):
        stack = []
        is_valid = True

        rotated = s[i:] + s[:i]
        for ch in rotated:
            if ch in match: # 여는 괄호
                stack.append(ch)
            else:           # 닫는 괄호
                if stack and match[stack[-1]] == ch:
                    stack.pop()
                else:
                    is_valid = False
                    break

        if is_valid and not stack:
            answer += 1

    return answer
